daemon/start.go

```diff
@@ -161,6 +161,13 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 		checkpointDir = container.CheckpointDir()
 	}
 
+	if runtime.GOOS == "solaris" {
```

Comment:
Currently on Linux this just fails when we try to actually run the image, wouldn't the same happen on `Solaris`?

Also, this probably make more sense in the `containerCreate` side of things, might as well fails as early as possible